transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1118,6 +1142,20 @@ private void callHandlerCallbackLater(AbstractChannelHandlerContext ctx, boolean
         }
     }
 
+    private void wakeUpForAutoFlush() {
+        if (autoFlush && ENQUEUE_WAKEUP_TASK_UPDATER.compareAndSet(this, 0, 1)) {
+            if (null == wakeupTask) {
+                wakeupTask = new Runnable() {
+                    @Override
+                    public void run() {
+                        ENQUEUE_WAKEUP_TASK_UPDATER.set(DefaultChannelPipeline.this, 0);
+                    }
+                };
+            }
+            channel().eventLoop().execute(wakeupTask);
+        }
+    }
+
```

Comment:
Hello! I've been watching this pull request with interest, and am really excited to see it moving forward.

It turns out we actually have [an application](https://github.com/relayrides/pushy) that fits this pattern really well (unpredictable writes from outside of the event loop and an HTTP/2 pipeline) and has [a benchmark](https://github.com/relayrides/pushy/pull/321) ready to go. I recognize you'll want something that actually lives in the Netty codebase at the end of the day, but if you want some preliminary results in the meantime, I'd be happy to run our benchmark against this branch.